.github/spam_cleaner.py

```diff
@@ -0,0 +1,458 @@
+import binascii
+import datetime
+import json
+import logging
+import os
+import pickle
+import re
+import smtplib
+
+from collections import defaultdict
+from urllib.parse import urlparse
+
+import pandas as pd
+
+from redminelib import Redmine
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.common.exceptions import NoSuchElementException
+
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+REDMINE_PAGE = "https://pulp.plan.io"
+
+REDMINE_LOGIN_PAGE = "https://pulp.plan.io/login"
+REDMINE_LOGIN_FORM_USERNAME = "pulpbot"
+REDMINE_LOGIN_FORM_PASSWORD = os.getenv("REDMINE_PASSWORD")
+
+REDMINE_USERS_PAGE = "https://pulp.plan.io/users"
+
+GOOGLE_CLIENT_CONFIG = os.getenv("GOOGLE_CLIENT_CONFIG")
+SPREADSHEET_ID = os.getenv("SPREADSHEET_ID")
+USERS_TO_VERIFY_SHEET_RANGE = "A1:AA1000"
+VERIFIED_USERS_SHEET_RANGE = "verified!A1:AA1000"
+SCOPES = ["https://www.googleapis.com/auth/spreadsheets"]
+
+EMAIL_USERNAME = "pulpbotnoreply@gmail.com"
+EMAIL_PASSWORD = os.getenv("EMAIL_PASSWORD")
+
+# https://github.com/actions/virtual-environments/pull/269
+CHROME_DRIVER_PATH = os.path.join(os.getenv("CHROMEWEBDRIVER"), "chromedriver")
+
+# https://stackoverflow.com/a/48689681
+URL_REGEX = (
+    r"((http|https)\:\/\/)[a-zA-Z0-9\.\/\?\:@\-_=#]+"
+    r"\.([a-zA-Z]){2,6}([a-zA-Z0-9\.\&\/\?\:@\-_=#])*"
+)
+
+TRUSTED_EMAIL_ADDRESS_DOMAINS = ["redhat.com", "atix.de"]
+TRUSTED_LINKS_DOMAINS = [
+    "github.com",
+    "gist.github.com",
+    "pulp.plan.io",
+    "travis-ci.com",
+    "pulpproject.org",
+    "docs.pulpproject.org",
+    "pulp-container.readthedocs.io",
+    "pulp-rpm.readthedocs.io",
+    "pulp-file.readthedocs.io",
+    "repos.fedorapeople.org",
+]
+
+REMOVAL_TIME_DELTA = datetime.timedelta(days=2)
+ISSUES_TO_CHECK_TIME_DELTA = datetime.timedelta(days=2)
```

Comment:
Let's make this value a very long time for the first time we run this. Maybe 3 years?